heron/schedulers/src/java/com/twitter/heron/scheduler/aurora/AuroraScheduler.java

```diff
@@ -183,8 +185,23 @@ public boolean onUpdate(Scheduler.UpdateTopologyRequest request) {
   }
 
   @Override
-  public void addContainers(Set<PackingPlan.ContainerPlan> containersToAdd) {
-    controller.addContainers(containersToAdd.size());
+  public Set<PackingPlan.ContainerPlan> addContainers(
+      Set<PackingPlan.ContainerPlan> containersToAdd) {
+    // Do the actual containers adding
+    LinkedList<Integer> newAddedContainerIds = new LinkedList<>(
+        controller.addContainers(containersToAdd.size()));
+    Set<PackingPlan.ContainerPlan> remapping = new HashSet<>();
+    // Do the remapping:
+    // use the `newAddedContainerIds` to replace the container id in the `containersToAdd`
+    for (PackingPlan.ContainerPlan cp : containersToAdd) {
```

Comment:
I think it should be better to test containersToAdd.size() == newAddedContainerId.size().

The controller.addContainers() function could return empty set or smaller set, and we need to handle the two cases here.

Reply:
done